src/Avalonia.Controls/ItemsControl.cs

```diff
@@ -506,5 +507,21 @@ private void UpdatePseudoClasses(int itemCount)
 
             return null;
         }
+
+        (int Index, int? TotalCount) IChildIndexProvider.GetChildIndex(ILogical child)
+        {
+            if (Presenter is IChildIndexProvider innerProvider)
+            {
+                return innerProvider.GetChildIndex(child);
+            }
+
+            if (child is IControl control)
+            {
+                var index = ItemContainerGenerator.IndexFromContainer(control);
+                return (index, ItemCount);
+            }
+
+            return (-1, ItemCount);
+        }
```

Comment:
Call me old fashioned but returning two values in a public method looks bad to me. I also think the interface should be more generic along the lines of IList as discussed [here](https://github.com/AvaloniaUI/Avalonia/issues/6262#issuecomment-881505715). The 'child' terminology is growing on me though.

```csharp
public interface IListChildren
{
    int IndexOfChild();
    int ChildCount { get; }
}
```

Reply:
Ok, I just realized you are using `ItemCount`. That means it should probably go back to my original thinking and be much more general-purpose.

public interface IItemList
{
    int IndexOfItem(ILogical item);
    int ItemCount { get; }
}

Reply:
When I see "item" I think about items source and data context, model items, and not visual items (controls). For later usually "children" is used in xaml. And in this interface we actually need to get child index.

Reply:
On other hand, we literally talking about Nth-**Child**.
But in the same time, I prefer ChildrenCount or TotalCount as separated property, yes.

Reply:
Yes, my only point is this interface could apply in other situations its a listing interface.